### PR TITLE
Introduce network-specific standard scripts

### PR DESCRIPTION
**Pull request by zeptin**
This fix allows the correct OP_RETURN maximum length to be enforced by mempools on each network type. Note that this is not a consensus parameter and only affects standardness of the transaction.

Future work to be done: remove all usages of the static templates (e.g. in the wallet transaction handler), and create templates from the network instead, similar to the consensus factories.

Ultimately there should be as little standardness-checking within the NBitcoin namespace as possible.

See related issue:
https://github.com/stratisproject/NStratis/issues/13

### CODE REVIEW

**Comment by zeptin** [comment]
Looks like some cold staking tests are now failing, I'll fix that.

**Comment by zeptin** [comment]
Hmm, that shouldn't be the case, it should be getting the Stratis network value of 40 for cold staking. Where did you see that?

I do see the tweaking for cold staking has broken one or two low level tests though, I'll correct those.

File: src/NBitcoin/StandardScriptsRegistry.cs [hunk 1]
```diff
@@ -10,27 +10,36 @@ namespace NBitcoin
...
     public class StandardScriptsRegistry : IStandardScriptsRegistry
     {
-        public bool IsStandardTransaction(Transaction tx, Network network)
+        /// <summary>
+        /// Registers a new standard script template if it does not exist yet based on <see cref="ScriptTemplate.Type"/>.
+        /// </summary>
+        /// <param name="scriptTemplate">The standard script template to register.</param>
+        public virtual void RegisterStandardScriptTemplate(ScriptTemplate scriptTemplate)
+        {
+            StandardScripts.RegisterStandardScriptTemplate(scriptTemplate);
```
Comment (dangershony) [inline src/NBitcoin/StandardScriptsRegistry.cs]: Should we not delete the static filed now?

File: src/Stratis.Bitcoin.Features.ColdStaking/ColdStakingFeature.cs [hunk 2]
```diff
@@ -252,7 +252,7 @@ public static IFullNodeBuilder UseColdStakingWallet(this IFullNodeBuilder fullNo
                 throw new InvalidOperationException("Cold staking can only be used on a Stratis network.");
 
             // Register the cold staking script template.
-            StandardScripts.RegisterStandardScriptTemplate(ColdStakingScriptTemplate.Instance);
+            fullNodeBuilder.Network.StandardScriptsRegistry.RegisterStandardScriptTemplate(ColdStakingScriptTemplate.Instance);
```
Comment (dangershony) [inline src/Stratis.Bitcoin.Features.ColdStaking/ColdStakingFeature.cs]: I think we should register the script templates when we build the network

**Pull request merged**